src/pytest_memray/plugin.py

```diff
@@ -118,10 +140,28 @@ def pytest_unconfigure(self, config: Config) -> Generator[None, None, None]:
         if os.environ.get("MEMRAY_RESULT_PATH"):
             del os.environ["MEMRAY_RESULT_PATH"]
 
+    def _default_runner(
+        self, func: Callable[[], Any], results_file: Path, *args: Any, **kwargs: Any
+    ) -> Any:
+        result = None
+        with Tracker(results_file):
+            result = func()
+        return result
+
     @hookimpl(hookwrapper=True)  # type: ignore[misc] # Untyped decorator
-    def pytest_pyfunc_call(self, pyfuncitem: Function) -> object | None:
+    def pytest_pyfunc_call(self, pyfuncitem: Function) -> Generator[None, None, None]:
         func = pyfuncitem.obj
 
+        runner: Runner = self._default_runner
+        marker_args: Tuple[Any, ...] = tuple()
+        marker_kwargs: Mapping[str, Any] = dict()
+        for marker in pyfuncitem.iter_markers():
+            if marker.name in MARKERS_RUNNERS:
+                runner = MARKERS_RUNNERS[marker.name]
+                marker_args = marker.args
+                marker_kwargs = marker.kwargs
+                break
```

Comment:
If multiple markers are applied to one test, we're silently ignoring all but the first one we find, I believe. I'd suggest we should raise an exception instead. Otherwise  code like:
```py
import pytest

pytestmark = pytest.mark.limit_memory("1 MB")

def test_func1():
    ...

def test_func2():
    ...

@pytest.mark.check_leaks("1 KB")
def test_func3():
    ...
```

will accidentally only have one of the two checks that they thought they were enabling.